### PR TITLE
Add multi metric leaderboard

### PR DESCRIPTION
**Pull request by Ram81**
### Description

- [x] Add boolean flag in challenge model to enable leaderboard for all metrics
- [x] Add leaderboard API changes to use `order_by` field to fetch the leaderboard if specified otherwise use `default_order_by`
- [x] Add frontend changes to show dropdown and generate separate URL for each metric

### CODE REVIEW

File: apps/challenges/serializers.py [hunk 1]
```diff
@@ -153,8 +155,13 @@ class Meta:
             "visibility",
             "show_leaderboard_by_latest_submission",
             "show_execution_time",
+            "leaderboard_schema",
         )
 
+    def get_leaderboard_schema(self, obj):
+        print(obj.leaderboard.schema)
```
Comment (RishabhJain2018) [inline apps/challenges/serializers.py:162]: Please remove the `print` statement?

**Review by RishabhJain2018 — CHANGES_REQUESTED** [review]
@Ram81 Do you have a working gif of it?

File: apps/challenges/models.py [hunk 2]
```diff
@@ -172,6 +172,8 @@ def __init__(self, *args, **kwargs):
     desired_worker_instance = models.IntegerField(
         null=True, blank=True, default=1
     )
+    # Allow ordering leaderboard by all metrics
+    is_multi_metric_leaderboard = models.BooleanField(default=False)
```
Comment (RishabhJain2018) [inline apps/challenges/models.py:176]: Since we have the `metadata` field in the database, I don't think we will need this field.

**Review by RishabhJain2018 — CHANGES_REQUESTED** [review]

**Comment by RishabhJain2018** [comment]
@Ram81 Can you please resolve the conflicts here?

**Review by RishabhJain2018 — COMMENTED** [review]
LGTM :), Is there a screenshot of the UI?

**Comment by RishabhJain2018** [comment]
@Ram81

**Comment by Ram81** [comment]
Screen Shot 2021-08-09 at 11 25 06 AM


Here are the screenshots
Screen Shot 2021-08-09 at 11 25 54 AM

**Comment by RishabhJain2018** [comment]
Can we please bold the entries by which it is ordered?

**Comment by RishabhJain2018** [comment]
@Ram81 Is this PR ready to merge?

**Comment by Ram81** [comment]
Screenshot from 2021-08-17 21-58-04

@RishabhJain2018 this PR is ready for review. Attached a screenshot that shows a leaderboard with selected order by field column being highlighted

**Review by RishabhJain2018 — APPROVED** [review]

**Pull request merged**